packages/provider/wrapper.js

```diff
@@ -32,6 +64,13 @@ module.exports = {
       const originalSendAsync = provider.sendAsync.bind(provider);
       provider.sendAsync = this.send(originalSendAsync, preHook, postHook);
     }
+
+    // EIP-1193 provider wrapping
+    if (provider.request === undefined && provider.send !== undefined) {
+      provider.request = make1193ProviderRequest(provider);
+    }
+    provider.request = provider.request.bind(provider);
```

Comment:
Hm, is this line necessary uncondtionally?  It seems odd to include in the case where `provider.request` is already defined.  Should this be moved inside the conditional?  If so, perhaps `make1193ProviderRequest` could be modified to return `modifiedRequest.bind(provider)`, rather than setting it in two steps (it's nice to avoid mutation when it's reasonable to do so :) ).